spec/models/manageiq/providers/embedded_terraform/automation_manager/stack_spec.rb

```diff
@@ -30,4 +30,125 @@
       end
     end
   end
+
+  describe "#raw_stdout" do
+    let(:stack) { FactoryBot.create(:terraform_stack) }
+    let(:template) { FactoryBot.create(:terraform_template) }
+
+    context "when miq_task.job present" do
+      require 'webmock/rspec'
+      require 'json'
+      let(:terraform_runner_url) { "https://1.2.3.4:7000" }
+      let(:hello_world_retrieve_response) do
+        JSON.parse(File.read(File.join(__dir__, "../../../../../lib/terraform/runner/data/responses/hello-world-retrieve-success.json")))
+      end
+      let(:miq_task) do
+        miq_task = FactoryBot.create(:miq_task)
+        miq_task.job = job
+        miq_task
+      end
+
+      let(:job) do
+        ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Job.create_job(template, {}, {}, []).tap do |job|
+          job.state = "finished"
+          job.options = {
+            :terraform_stack_id => hello_world_retrieve_response['stack_id']
+          }
+        end
+      end
+
+      let(:terraform_runner_stdout) { hello_world_retrieve_response['message'] }
+      let(:terraform_runner_stdout_html) { TerminalToHtml.render(terraform_runner_stdout) }
+
+      retrieve_stub = nil
+
+      before do
+        stack.miq_task = miq_task
+
+        stub_const("ENV", ENV.to_h.merge("TERRAFORM_RUNNER_URL" => terraform_runner_url))
+
+        retrieve_stub = stub_request(:post, "#{terraform_runner_url}/api/stack/retrieve")
+                        .with(:body => hash_including({:stack_id => hello_world_retrieve_response['stack_id']}))
+                        .to_return(
+                          :status => 200,
+                          :body   => hello_world_retrieve_response.to_json
+                        )
```

Comment:
Oh you are trying to get access to this output from outside the before block, I'm not sure we really care about the have_been_requested.times(1) if the output is correct that should be sufficient so you could just drop the `retrieve_stub =` part